content/advent-2016/sea-of-open-source.md

```diff
@@ -0,0 +1,165 @@
+other or some other language construct.
+
+There are a few guidelines for [code review](https://github.com/golang/go/wiki/CodeReviewComments) that you can read to
+help out with the review. Furthermore, [Effective Go](https://golang.org/doc/effective_go.html) and articles like
+[Idiomatic Go](https://dmitri.shuralyov.com/idiomatic-go), [Avoid gotchas](https://divan.github.io/posts/avoid_gotchas) or 
+[Gotchas and common mistakes in Go](http://devs.cloudimmunity.com/gotchas-and-common-mistakes-in-go-golang/) can help you
+quickly understand what are the common patters you should look out for.
+
+A quick way to check some of these is to have a look at the package using a
+more automated way. First and the simplest is to run [Go Report Card](https://goreportcard.com/) and see
+what problems does the package have. Things like `gofmt` should be a must for
+any package out there so that quickly can help identify packages which are not
+correctly formatted.
+
+If Go Report Card checks out ok, it's time to download the package locally and
+run [Go Meta Linter](https://github.com/alecthomas/gometalinter) against it.
+That will quickly highlight any problems with the package. Hopefully this will
+not discover anything but if that happens then I can at least open an issue so
+that the author is informed about this. If I like how the code looks like and
+I decide to use the package, now I know I can also quickly contribute to it.
+
+Placing your full trust in automated tools however might also lead you down the
+wrong path as perfect valid code, that can pass checks, can also be not
+idiomatic. That's why it's good to use these tools but always have a look at
+what's happening under the hood for yourself.
+
+An important property of Go packages is how well they work with other packages
+so if you want to have a single takeaway from this article then this would be:
+Go as a language encourages composition and this can be observed throughout the
+standard library. Many of the popular packages in the open-source community
+also respect and embrace this property and allow you to quickly combine them
+in order to achieve the powerful functionality you need.
+
+Finally, there's a shortcut you can take, tho I have to admit, it involves a
+lot of trust on people on the Internet. Ask about the package you are looking
+at in places like the [Gophers Slack](https://gophers.slack.com/), you can get
+an [invite here](https://invite.slack.golangbridge.org/), the [Go Forum](https://forum.golangbridge.org/), 
+or [golang-nuts mailing list](https://groups.google.com/forum/#!forum/golang-nuts).
+
+But what should you do if you cannot find a good package or any package at all?
+
+If you find a package that maybe doesn't cover everything that you need or
+the quality it's not up on par, I highly suggest contributing to it. 
+
+You can read in [this article](http://blog.sgmansfield.com/2016/06/working-with-forks-in-go/) how to work with forks for
```

Comment:
https://splice.com/blog/contributing-open-source-git-repositories-go/ is also good